Replicator/tests/ReplicatorAPITest.cc

```diff
@@ -419,3 +427,79 @@ TEST_CASE_METHOD(ReplicatorAPITest, "Is Document Pending", "[Push]") {
     CHECK(err.code == 0);
 }
 #endif
+
+#ifdef COUCHBASE_ENTERPRISE
+TEST_CASE_METHOD(ReplicatorAPITest, "Rapid Restarts", "[C][Push][Pull]") {
+    importJSONLines(sFixturesDir + "names_100.json");
+
+    createDB2();
+    _mayGoOffline = true;
+    C4Error err;
+    REQUIRE(startReplicator(kC4Continuous, kC4Continuous, &err));
+    waitForStatus(_repl, kC4Busy, 50);
+    
+    C4ReplicatorActivityLevel expected = kC4Stopped;
+    SECTION("Stop / Start") {
+        c4repl_stop(_repl);
+        c4repl_start(_repl);
+        expected = kC4Idle;
+    }
+
+    SECTION("Stop / Start / Stop") {
+        c4repl_stop(_repl);
+        c4repl_start(_repl);
+        c4repl_stop(_repl);
+    }
+
+    SECTION("Suspend / Unsuspend") {
+        c4repl_setSuspended(_repl, true);
+        c4repl_setSuspended(_repl, false);
+        expected = kC4Idle;
+    }
+    
+    SECTION("Suspend / Unsuspend / Suspend") {
+        c4repl_setSuspended(_repl, true);
+        c4repl_setSuspended(_repl, false);
+        c4repl_setSuspended(_repl, true);
+        expected = kC4Offline;
+    }
+    
+    SECTION("Stop / Suspend") {
+        c4repl_stop(_repl);
+        c4repl_setSuspended(_repl, true);
+    }
+    
+    SECTION("Suspend / Stop") {
+        c4repl_setSuspended(_repl, true);
+        c4repl_stop(_repl);
```

Comment:
Just curious if it needs to set expected to kC4Stopped?

Reply:
That's the default on line 441 `C4ReplicatorActivityLevel expected = kC4Stopped;`